### PR TITLE
Format assignments more consistently

### PR DESCRIPTION
**Pull request by thorn0**
## Description

A take on #2482

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — APPROVED** [review]

**Review by duailibe — APPROVED** [review]

File: tests/flow-repo/declare_export/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1327,8 +1327,8 @@ var numberValue6 = require("ES6_ExportFrom_Intermediary2").numberValue1;
 var ap1: number = numberValue6;
 var ap2: string = numberValue6; // Error: number ~> string
 
-var numberValue2_renamed2 = require("ES6_ExportFrom_Intermediary2")
-  .numberValue2_renamed2;
+var numberValue2_renamed2 =
+  require("ES6_ExportFrom_Intermediary2").numberValue2_renamed2;
```
Comment (alexander-akait) [inline tests/flow-repo/declare_export/__snapshots__/jsfmt.spec.js.snap:1331]: I am afraid we will get a lot of issues on this

Comment (thorn0) [reply tests/flow-repo/declare_export/__snapshots__/jsfmt.spec.js.snap:1331]: Do you think many people prefer to have that property on a separate line? We'll see.

I'm done with this PR, BTW. What do you think about the rest of the changes? If you approve it (or somebody else), I'll merge it today.

**Review by lipis — APPROVED** [review]

File: tests/js/ternaries/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -892,8 +892,10 @@ a
 																),
 																a()
 																	? {
-																			a: a(),
-																			b: [],
+																			a:
+																				a(),
+																			b:
+																				[],
```
Comment (xixixao) [inline tests/js/ternaries/__snapshots__/jsfmt.spec.js.snap:898]: These look broken?

File: tests/js/comments-closure-typecast/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -372,9 +478,11 @@ foo = /** @type {!Foo} */ (/** @type {!Baz} */ (baz).bar);
...
 const BarImpl = /** @type {BarConstructor} */ (
   /** @type {unknown} */
-  (function Bar() {
-    throw new Error("Internal error: Illegal constructor");
-  })
+  (
+    function Bar() {
+      throw new Error("Internal error: Illegal constructor");
+    }
+  )
```
Comment (xixixao) [inline tests/js/comments-closure-typecast/__snapshots__/jsfmt.spec.js.snap:485]: minor but also looks unintended?

File: tests/js/ternaries/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (thorn0) [reply tests/js/ternaries/__snapshots__/jsfmt.spec.js.snap:898]: It's beyond `printWidth`. If the names of the keys were longer, it wouldn't look that bad. Special handling for short keys is a separate open issue: https://github.com/prettier/prettier/issues/3711

File: tests/js/comments-closure-typecast/__snapshots__/jsfmt.spec.js.snap (hunk 3 above)
Comment (thorn0) [reply tests/js/comments-closure-typecast/__snapshots__/jsfmt.spec.js.snap:485]: Something to improve in the future. The idea was to format type assertions similarly to function calls while preserving at the same the usual behavior of comments in Prettier. If a comment is on its own line, Prettier doesn't change that. But now I see that the behavior of the parens could depend on this condition (whether the comment is own-line) too.

**Pull request merged**